From d04257360c5b5cafbfdb67a3d1bcaf270f101465 Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Fri, 3 Mar 2023 20:06:09 +0900
Subject: [PATCH] tools: explicitly specify "setup" subcommand

As invoking meson without subcommand is deprecated since 0.64.0.

(cherry picked from commit e3b2f7c056551fa50ba6ed13703b8e63ba8303c7)

Related: RHEL-30372
---
 tools/meson-build.sh | 2 +-
 tools/oss-fuzz.sh    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/meson-build.sh b/tools/meson-build.sh
index c8370a7e6a..2ef60cfa8e 100755
--- a/tools/meson-build.sh
+++ b/tools/meson-build.sh
@@ -10,7 +10,7 @@ CC="$5"
 CXX="$6"
 
 # shellcheck disable=SC2086
-[ -f "$dst/build.ninja" ] || CC="$CC" CXX="$CXX" meson "$src" "$dst" $options
+[ -f "$dst/build.ninja" ] || CC="$CC" CXX="$CXX" meson setup "$src" "$dst" $options
 
 # Locate ninja binary, on CentOS 7 it is called ninja-build, so
 # use that name if available.
diff --git a/tools/oss-fuzz.sh b/tools/oss-fuzz.sh
index 7e9312b833..2e64475c6d 100755
--- a/tools/oss-fuzz.sh
+++ b/tools/oss-fuzz.sh
@@ -73,7 +73,7 @@ else
     fi
 fi
 
-if ! meson "$build" "-D$fuzzflag" -Db_lundef=false; then
+if ! meson setup "$build" "-D$fuzzflag" -Db_lundef=false; then
     cat "$build/meson-logs/meson-log.txt"
     exit 1
 fi
